data/process_offline_test_data.py

```diff
@@ -0,0 +1,141 @@
+import json
+from datetime import datetime
+from uuid import UUID, uuid4
+
+import pandas as pd
+from tqdm import tqdm
+
+from poprox_concepts import Article, ClickHistory
+
+
+def custom_encoder(obj):
+    if isinstance(obj, UUID):
+        return str(obj)
+    elif isinstance(obj, datetime):
+        return obj.isoformat()
+
+
+def load_data():
+    val_behavior_df = pd.read_table(
+        "./val_mind_large/behaviors.tsv",
+        header="infer",
+        usecols=range(5),
+        names=["impression_id", "user", "time", "clicked_news", "impressions"],
+    )
+
+    val_behavior_df.fillna("", inplace=True)
```

Comment:
In the interest of "minimal migration of existing logic", let's not change this yet, but going forward: I consider blanket `fillna` to be an anti-pattern — we should handle NAs in a targeted and field-appropriate manner.